python/ray/dag/input_node.py

```diff
@@ -57,19 +58,29 @@ class InputNode(DAGNode):
     ...     # dag_input["m1"], dag_input["m2"]
     """
 
-    def __init__(self, *args, _other_args_to_resolve=None, **kwargs):
+    def __init__(
+        self,
+        *args,
+        input_types: Dict[Union[int, str], type] = None,
```

Comment:
This should be Union[type, Dict[...]] because in common case input attribute is not used.

Reply:
Changed it to `Optional[Dict[Union[int, str], type]]`. Not sure why it is `Union[type, Dict[...]]` instead of just `Dict[...]`?

Reply:
Because user can specify `input_types=int` with the input node literally just the int instead of access it through attribute right?